### PR TITLE
[simd/v3i]: Implement v128 store_lane

### PR DESCRIPTION
**Pull request by haoyu-zc**
Tested by:
```
make v3i -j && bin/spectest.v3i -tk test/regress/simd/simd_store64_lane.bin.wast
make v3i -j && bin/spectest.v3i -tk test/regress/simd/simd_store32_lane.bin.wast
make v3i -j && bin/spectest.v3i -tk test/regress/simd/simd_store16_lane.bin.wast
make v3i -j && bin/spectest.v3i -tk test/regress/simd/simd_store8_lane.bin.wast
```

### CODE REVIEW

File: src/engine/v3/V3Interpreter.v3 [hunk 1]
```diff
@@ -1196,6 +1200,32 @@ component V3Interpreter {
...
+		def half_lanes = 8 >> log2_size;
+		if (idx < half_lanes) { // Extract a lane from low
+			var shift = u6.view(idx << u3.+(log2_size, 3));
+			val = view(low >> shift);
+		} else { // Extract a lane from high
+			var shift = u6.view((idx - half_lanes) << u3.+(log2_size, 3));
+			val = view(high >> shift);
+		}
+		// Write to memory
+		var t = memory.range_oil_32(u32.!(memarg.offset), u32.!(index), size);
```
Comment (titzer) [inline src/engine/v3/V3Interpreter.v3:1225]: You should be able to use `memory.range_oil_64` here and avoid the `u32` checks above. Then this routine will be ready for the `memory64` wasm extension.

Comment (haoyu-zc) [reply src/engine/v3/V3Interpreter.v3:1225]: Then I guess these two checks could be also removed?
```
		if (!u32.?(index)) return trap(TrapReason.MEM_OUT_OF_BOUNDS);
		if (!u32.?(memarg.offset)) return trap(TrapReason.MEM_OUT_OF_BOUNDS);
```

Comment (titzer) [reply src/engine/v3/V3Interpreter.v3:1225]: Yep.

**Review by titzer — APPROVED** [review]
lgtm

**Pull request merged**